sql/wsrep_mysqld.cc

```diff
@@ -488,6 +488,7 @@ int wsrep_init()
       WSREP_ERROR("wsrep_load() failed to load the provider('%s'): %s (%d). Reverting to no provider.",
                   wsrep_provider, strerror(rcode), rcode);
 
+      wsrep_provider_set = false;
```

Comment:
I am not sure, I think we should also set `wsrep_provider_set = false` in case if `wsrep_load()` is successful and  `wsrep_provider` is set "none"/empty?

i.e, Inside this block https://github.com/percona/percona-xtradb-cluster/blob/ccd2185ba1247b3460f6074a40ec7582ccbf4f61/sql/wsrep_mysqld.cc#L515-L518

Reply:
That's a good question. For WSREP_NONE, that's already done in mysqld.cc. For the empty string... there are a few places where wsrep_mysqld.cc checks that, but this condition wasn't included in the WSREP_ON macro previously, so I'm not sure.

Reply:
When the server is started with `wsrep_provider=""`, `wsrep_load()` fails and we enter this block https://github.com/percona/percona-xtradb-cluster/blob/ccd2185ba1247b3460f6074a40ec7582ccbf4f61/sql/wsrep_mysqld.cc#L491-L499

Here, we recursively call `wsrep_init()` with `wsrep_provider=none` and this second call to the `wsrep_init()` will be successful.

Considering this flow(with the proposed solution), we are setting `wsrep_provider_set = false` only when the server is started with `wsrep_provider=""` and it is not set if the server is started with `wsrep_provider=none`.

I think it is probably better to move `wsrep_provider_set = false` to this block, so that we set it correctly for both none and empty cases. https://github.com/percona/percona-xtradb-cluster/blob/ccd2185ba1247b3460f6074a40ec7582ccbf4f61/sql/wsrep_mysqld.cc#L518-L525

Reply:
Done.